### PR TITLE
Feature/dont limit import

### PR DESCRIPTION
**Pull request by jbrw1984**
Pretty big update, adding create/updated flag. Also upping to 1.5.0

### CODE REVIEW

File: airports.js [hunk 1]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
   });
 }
 
+function regenerateCsvAirports(callback) {
+  rawAirports = fs.readFileSync(config.airports.csvInPath, { aencoding: 'utf8' });
```
Comment (dylanfoster) [inline airports.js]: `encoding`

File: airports.js [hunk 2]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+          var fullAddress = airportRow['city'] + ', ' + airportRow['state'] + ' ' + airportRow['country'];
+        } else {
+          var fullAddress = airportRow['city'] + ', ' + airportRow['country'];
+        }
+        var fullUrl = config.geocoder.baseUrl + encodeURIComponent(fullAddress) + config.geocoder.keyUrl;
+        request(fullUrl, function (error, response, bodyString) {
+          body = JSON.parse(bodyString);
+          var newStateCode = '';
+          var newCountryCode = '';
+          if (body.results[0] != undefined) {
```
Comment (dylanfoster) [inline airports.js]: `if(body.results && body.results.length)` is probably safer

File: airports.js [hunk 3]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+            for (var component of body.results[0].address_components) {
+              // country -> country
+              // administrative_area_level_1 -> state
+              if (component.types.indexOf('country') !== -1) {
+                if (component.short_name != undefined) {
```
Comment (dylanfoster) [inline airports.js]: `if (component.short_name`

File: airports.js [hunk 4]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+          airportRow.stateCode = newStateCode;
+          airportRow.countryCode = newCountryCode;
+          allAirportRows.push(_.values(airportRow));
+          setTimeout(cb, config.geocoder.delay);
+        });
+      },
+      function (err) {
+        allAirportRows = _.orderBy(allAirportRows, ['code'], ['asc']);
+        csv.stringify(allAirportRows, function (err, data) {
+          fs.writeFileSync(config.airports.csvOutPath, data, { encoding: 'utf8' });
```
Comment (dylanfoster) [inline airports.js]: Catch err

File: airports.js [hunk 5]
```diff
@@ -392,14 +440,21 @@ function generateMultiAirportCityCodes(callback) {
 
 if (require.main == module) {
   if (process.argv.length == 3) {
-    if (process.argv[2] === 'rail') {
-      regenerateCsv(function (err) {
+    if (process.argv[2] === 'regenRail') {
```
Comment (dylanfoster) [inline airports.js]: Normally I’d vote against this since it’s an unnecessary breaking change upstream

File: package.json [hunk 6]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "@travelbank/airports",
-  "version": "1.4.3",
+  "version": "1.5.1",
```
Comment (dylanfoster) [inline package.json]: Why 1.5.1?

File: airports.js (hunk 3 above)
Comment (dylanfoster) [reply airports.js]: Could probably also just include this in the check above

File: airports.js [hunk 7]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+    assert.equal(null, err);
+    // Using the limit to throttle request rate so as not to exceed our query limit to google
+    async.eachLimit(airportData, config.geocoder.limit, function (airportRow, cb) {
+        if (airportRow['country'] == 'United States') {
+          var fullAddress = airportRow['city'] + ', ' + airportRow['state'] + ' ' + airportRow['country'];
+        } else {
+          var fullAddress = airportRow['city'] + ', ' + airportRow['country'];
+        }
+        var fullUrl = config.geocoder.baseUrl + encodeURIComponent(fullAddress) + config.geocoder.keyUrl;
+        request(fullUrl, function (error, response, bodyString) {
```
Comment (dylanfoster) [inline airports.js]: Error check

File: airports.js [hunk 8]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+  allAirportRows.push(config.airports.headerRow);
+  console.log('Re-geocoding airport stations...');
+  csv.parse(rawAirports, {
+    columns: true
+  }, function (err, airportData) {
+    assert.equal(null, err);
+    // Using the limit to throttle request rate so as not to exceed our query limit to google
+    async.eachLimit(airportData, config.geocoder.limit, function (airportRow, cb) {
+        if (airportRow['country'] == 'United States') {
+          var fullAddress = airportRow['city'] + ', ' + airportRow['state'] + ' ' + airportRow['country'];
```
Comment (dylanfoster) [inline airports.js]: Dot notation for these

File: airports.js [hunk 9]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
...
+            for (var component of body.results[0].address_components) {
+              // country -> country
+              // administrative_area_level_1 -> state
+              if (component.types.indexOf('country') !== -1) {
+                if (component.short_name != undefined) {
+                  newCountryCode = component.short_name;
+                }
+              }
+              if (component.types.indexOf('administrative_area_level_1') !== -1) {
+                if (component.short_name != undefined && component.short_name !== component.long_name) {
```
Comment (szantogeri) [inline airports.js]: Is null a good value? If not use ```if (!component.short_name && ``` instead to avoid null as well

File: airports.js [hunk 10]
```diff
@@ -262,6 +258,58 @@ function regenerateCsv(callback) {
   });
 }
 
+function regenerateCsvAirports(callback) {
```
Comment (szantogeri) [inline airports.js]: This callback is never called in the function. Why we passing it then?

File: airports.js (hunk 9 above)
Comment (szantogeri) [inline airports.js]: Dylan mentioned this already. Is null a good value here? If not use 
```if (!component.short_name &&``` instead to avoid null as well

**Comment by jbrw1984** [comment]
Spoke w/ @dylanfoster - this is more of a maintenance script, not worth the time to clean up right now. Merging. :rage4:

**Pull request merged**